azurelinuxagent/agent.py

```diff
@@ -303,13 +304,23 @@ def _validate_log_collector_cgroup_slice():
     def setup_firewall(endpoint):
         logger.set_prefix("Firewall")
         threading.current_thread().name = "Firewall"
-        event.info(event.WALAEventOperation.Firewall, "Setting up firewall after boot. Endpoint: {0}", ustr(endpoint))
+        logger.info("Setting up firewall during boot. Endpoint: {0}", ustr(endpoint))
+
+        try:
+            run_command(['systemctl', 'is-enabled', '--type=service', 'firewalld.service']).rstrip()
+            logger.info("Firewalld is enabled. Will not setup the firewall rules to avoid conflicts.")
+            sys.exit(0)
+        except CommandError:
+            # Differences across versions of systemd make hard to determine whether the command failed because firewalld is not installed
+            # or for another reason. Assume it is not installed and continue.
+            pass
+
         try:
             firewall_manager = FirewallManager.create(endpoint)
             firewall_manager.setup()
-            event.info(event.WALAEventOperation.Firewall, "Successfully set the firewall rules")
+            logger.info("Successfully set up the firewall rules:\n{0}", firewall_manager.get_state())
         except Exception as error:
-            event.error(event.WALAEventOperation.Firewall, "Unable to add firewall rules. Error: {0}", ustr(error))
+            logger.warn("Unable to add firewall rules. Error: {0}", ustr(error))
```

Comment:
can we add current state when error happens? It helps if setup left undesirable state.

Reply:
thanks; updated